src/main/java/seedu/address/model/itemtype/TypeOfItem.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.itemtype;
+
+import static java.util.Objects.requireNonNull;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the type of item in the order book.
+ * Guarantees: immutable; item is valid as declared in {@link #isValidItemName(String)}}
+ */
+
+public class TypeOfItem {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Type Of Item should be alphanumeric";
+
+    /**
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+
+    public final String itemType;
+
+    /**
+     * Constructs a {@code Name}.
+     *
+     * @param itemType A valid monetary value.
+     */
+    public TypeOfItem(String itemType) {
+        requireNonNull(itemType);
+        checkArgument(isValidItemName(itemType), MESSAGE_CONSTRAINTS);
+        this.itemType = itemType;
+    }
+
+    /**
+     * Returns true if a given string is a valid transaction ID.
+     */
+    public static boolean isValidItemName(String test) {
```

Comment:
This sound confusing to me. I think you should change to isValidItemType, furthermore do update the java doc at the start of the class if changed.